File: baekjoon/stage16/1003_피보나치_함수.py
import sys

def result(n):
    zero_count = [1, 0]
    one_count = [0, 1]
    if n <= 1:
        return zero_count, one_count
    for i in range(2, n+1):
        zero_count.append(zero_count[i-1]+zero_count[i-2])
        one_count.append(one_count[i-1]+one_count[i-2])
    return zero_count, one_count

t = int(sys.stdin.readline())

for i in range(0, t):
    n = int(sys.stdin.readline())
    zero_count, one_count = result(n)
    print('%d %d' %(zero_count[n], one_count[n]))

File: baekjoon/stage16/1003_피보나치_함수.go
package main

import (
	"os"
	"fmt"
	"bufio"
)

func main() {
	var t int
	reader := bufio.NewReader(os.Stdin)
	fmt.Fscanln(reader, &t)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	for i := 0 ; i < t; i++ {
		var n int
		fmt.Fscanln(reader, &n)
		zeroCount, oneCount := result(n)
		fmt.Fprintf(writer, "%d %d\n", zeroCount[n], oneCount[n])
	}
}

func result(n int) (zeroCount, oneCount[]int) {
	zeroCount = []int{1, 0}
	oneCount = []int{0, 1}

	if n <= 1 {
		return
	}
	for i := 2; i< n+1; i++ {
		zeroCount = append(zeroCount, zeroCount[i-1]+zeroCount[i-2])
		oneCount = append(oneCount, oneCount[i-1]+oneCount[i-2])
	}
	return
}